turbopack/crates/turbo-tasks/src/manager.rs

```diff
@@ -363,7 +368,7 @@ task_local! {
     /// The current TurboTasks instance
     static TURBO_TASKS: Arc<dyn TurboTasksApi>;
 
-    static CURRENT_TASK_STATE: RefCell<CurrentTaskState>;
+    static CURRENT_TASK_STATE: Arc<RwLock<CurrentTaskState>>;
```

Comment:
I think it would be cheaper to split the `CurrentTaskState` into immutable and mutable parts and only wrap the mutable parts in an `Arc<Mutex<...>>`. The immutable parts can be cloned, like `task_id`, `execution_id`, `function_meta`. Especially `task_id` is accessed pretty often and won't require a lock.

Reply:
Given that I ran CPU benchmarks on this PR and found no measurable difference, let's leave this as-is for now, but we can consider pulling out the immutable parts in the future.

#68963 does this to some degree, as `CurrentLocalTaskState` is an immutable object with no lock, but you have a good point about `task_id`, which that PR does not address.